### PR TITLE
714101: Fix for issue where shovels were not getting restarted

### PR DESCRIPTION
**Pull request by rorytorneymf**
https://internal.almoctane.com/ui/entity-navigation?p=131002/6001&entityType=work_item&id=714101

### CODE REVIEW

File: worker-message-prioritization-distribution/src/main/java/com/github/workerframework/workermessageprioritization/redistribution/shovel/NonRunningShovelChecker.java [hunk 1]
```diff
@@ -38,13 +38,15 @@ public final class NonRunningShovelChecker implements Runnable
...
     private final String rabbitMQVHost;
+    private final String rabbitMQUri;
     private final long nonRunningShovelTimeoutMilliseconds;
     private final long nonRunningShovelTimeoutCheckIntervalMilliseconds;
 
     public NonRunningShovelChecker(
             final RabbitManagementApi<ShovelsApi> shovelsApi,
             final LoadingCache<String,RabbitManagementApi<ShovelsApi>> nodeSpecificShovelsApiCache,
             final String rabbitMQVHost,
+            final String rabbitMQUri,
```
Comment (andyreidz) [inline worker-message-prioritization-distribution/src/main/java/com/github/workerframework/workermessageprioritization/redistribution/shovel/NonRunningShovelChecker.java:49]:
```suggestion
            final String rabbitAmpqUri,
```
Would this help differentiate it from the managment uri? If you agree rename elsewhere.

File: worker-message-prioritization-distribution/src/main/java/com/github/workerframework/workermessageprioritization/redistribution/shovel/ShovelRepairer.java [hunk 2]
```diff
@@ -133,40 +135,50 @@ private static boolean recreateShovel(
...
+        //   endpoint, so if we can't get the shovel from /api/parameters/shovel/{vhost}, we do not know what the src-delete-after or
+        //   ack-mode properties should be
+        // - Therefore, we are recreating the shovel with a src-delete-after of 0, so that the shovel gets recreated then immediately
+        //   deleted. This allows the shovel to be created as normal (if still required) during the next loop of the application.
+
         final Shovel shovel = new Shovel();
-        shovel.setSrcDeleteAfter(retrievedShovel.getSrcDeleteAfter());
-        shovel.setAckMode(retrievedShovel.getAckMode());
+        shovel.setSrcDeleteAfter(0);
+        shovel.setAckMode(ACK_MODE);
```
Comment (andyreidz) [inline worker-message-prioritization-distribution/src/main/java/com/github/workerframework/workermessageprioritization/redistribution/shovel/ShovelRepairer.java:153]: Given that some key parameters aren't available, we shouuld introduce a different data type to be returned from /api/shovels/{vhost}

**Review by andyreidz — CHANGES_REQUESTED** [review]

File: worker-message-prioritization-distribution/src/main/java/com/github/workerframework/workermessageprioritization/redistribution/shovel/ShovelRepairer.java (hunk 2 above)
Comment (andyreidz) [reply worker-message-prioritization-distribution/src/main/java/com/github/workerframework/workermessageprioritization/redistribution/shovel/ShovelRepairer.java:153]: The existing class could extend this new class and add the additional fields.

Comment (rorytorneymf) [reply worker-message-prioritization-distribution/src/main/java/com/github/workerframework/workermessageprioritization/redistribution/shovel/ShovelRepairer.java:153]: Any thoughts on what to rename these new classes to to make it clear what they should be used for?

Right now we have

- Shovel                                           (srcUri, destUri, srcQueue, destQueue, ackMode, srcDeleteAfter)
- RetrievedShovel extends Shovel   (ADDS: name, timestamp, state, node)


Should we change/rename these to:

- CommonShovelProps                           (srcUri, destUri, srcQueue, destQueue)
- GetShovelProps extends CommonShovelProps    (ADDS: name, timestamp, state, node)
- PutShovelProps extends CommonShovelProps    (ADDS: ackMode, srcDeleteAfter)

Comment (rorytorneymf) [reply worker-message-prioritization-distribution/src/main/java/com/github/workerframework/workermessageprioritization/redistribution/shovel/ShovelRepairer.java]: I've gone with these classes for now, can rename these if required:

- CommonShovelProps
- RetrievedShovel extends CommonShovelProps
- ShovelToCreate extends CommonShovelProps

Commit: https://github.com/WorkerFramework/worker-message-prioritization/pull/36/commits/59ce89806224d5b4e92f1fe6e86724e32cc20c6a

**Pull request merged**